app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/ApplicationController.scala

```diff
@@ -50,17 +52,50 @@ class ApplicationController @Inject() (
 
   def startSubscription(service: Service): Action[AnyContent] = ggAuthorisedUserWithEnrolmentsAction {
     implicit request => implicit loggedInUser: LoggedInUserWithEnrolments =>
-      enrolledForService(loggedInUser, service) match {
-        case Some(_) =>
+      try {
+        if (isUserEnrolledFor(loggedInUser, service)) throw SpecificEnrolmentExists(service)
+
+        loggedInUser.groupId match {
+          case Some(groupId) =>
+            hasGroupIdEnrolmentTo(groupId, service).flatMap { groupIdEnrolmentExists =>
+              if (groupIdEnrolmentExists) throw SpecificGroupIdEnrolmentExists(service)
+
+              cdsEnrolmentCheck(loggedInUser, groupId, service)
+            }
+          case None if enrolledCds(loggedInUser).isDefined =>
+            Future.successful(Redirect(routes.HasExistingEoriController.displayPage(service))) // AutoEnrolment
+          case None =>
+            Future.successful(Redirect(routes.EmailController.form(service, Journey.Subscribe))) // Whole journey
+        }
+      } catch {
+        case SpecificEnrolmentExists(service) =>
+          Future.successful(Redirect(routes.EnrolmentAlreadyExistsController.enrolmentAlreadyExists(service)))
+        case SpecificGroupIdEnrolmentExists(service) =>
+          // Below redirect should be to page that doesn't exists yet
```

Comment:
This is the new page with instructions on how you can add another user to your "group" enrolment?

Reply:
yes